cmd/eks-node-viewer/flag.go

```diff
@@ -0,0 +1,137 @@
+	"errors"
+	"flag"
+	"fmt"
+	"k8s.io/client-go/util/homedir"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+var (
+	homeDir    string
+	configPath string
+)
+
+func init() {
+	homeDir = homedir.HomeDir()
+	configPath = filepath.Join(homeDir, ".eks_node_viewer")
+}
+
+type Flags struct {
+	Context        string
+	NodeSelector   string
+	ExtraLabels    string
+	Kubeconfig     string
+	Resources      string
+	DisablePricing bool
+}
+
+func ParseFlags() (Flags, error) {
+	flagSet := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	var flags Flags
+
+	cfg, err := loadConfigFile()
+	if err != nil {
+		return Flags{}, fmt.Errorf("load config file: %w", err)
+	}
+
+	contextDefault := cfg.getValue("context", "")
+	flagSet.StringVar(&flags.Context, "context", contextDefault, "Name of the kubernetes context to use")
+
+	nodeSelectorDefault := cfg.getValue("nodeSelector", "")
+	flagSet.StringVar(&flags.NodeSelector, "nodeSelector", nodeSelectorDefault, "Node label selector used to filter nodes, if empty all nodes are selected ")
+
+	extraLabelsDefault := cfg.getValue("extra-labels", "")
+	flagSet.StringVar(&flags.ExtraLabels, "extra-labels", extraLabelsDefault, "A comma separated set of extra node labels to display")
+
+	// flag overrides env. var. and env. var. overrides config file
+	kubeconfigDefault := getStringEnv("KUBECONFIG", cfg.getValue("kubeconfig", filepath.Join(homeDir, ".kube", "config")))
+	flagSet.StringVar(&flags.Kubeconfig, "kubeconfig", kubeconfigDefault, "Absolute path to the kubeconfig file")
+
+	resourcesDefault := cfg.getValue("resources", "cpu")
+	flagSet.StringVar(&flags.Resources, "resources", resourcesDefault, "List of comma separated resources to monitor")
+
+	disablePricingDefault := cfg.getBoolValue("disable-pricing", false)
+	flagSet.BoolVar(&flags.DisablePricing, "disable-pricing", disablePricingDefault, "Disable pricing lookups")
+
+	if err := flagSet.Parse(os.Args[1:]); err != nil {
+		return Flags{}, err
+	}
+	return flags, nil
+}
+
+// --- env vars ---
+
+func getStringEnv(envName string, defaultValue string) string {
+	env, ok := os.LookupEnv(envName)
+	if !ok {
+		return defaultValue
+	}
+	return env
+}
+
+// --- config file ---
+
+type configFile map[string]string
+
+func (c configFile) getValue(key string, defaultValue string) string {
+	if val, ok := c[key]; ok {
+		return val
+	}
+	return defaultValue
+}
+
+func (c configFile) getBoolValue(key string, defaultValue bool) bool {
+	if val, ok := c[key]; ok {
+		if boolVal, err := strconv.ParseBool(val); err == nil {
+			return boolVal
+		}
+	}
+	return defaultValue
+}
+
+func loadConfigFile() (configFile, error) {
+	fileContent := make(map[string]string)
+	if _, err := os.Stat(configPath); errors.Is(err, os.ErrNotExist) {
+		return fileContent, nil
+	}
+
+	file, err := os.Open(configPath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		lineKV := strings.Split(scanner.Text(), "=")
```

Comment:
This will remove the need for strings.Join later and you can just check `if len(lineKV) == 2`

```suggestion
		lineKV := strings.SplitN(scanner.Text(), "=", 2)
```

Whenever I write this config file parsing code I usually also start with something like this to allow '#' comments:
```
line := strings.TrimSpace(scanner.Text());
if strings.HasPrefix(line,"#") {
  continue;
}
```

Reply:
Thanks @tzneal, updated the PR with your suggestions